### PR TITLE
recover_cells_and_kzg_proofs(): Don't compute individual proofs. Always do FK20.

### PR DESCRIPTION
**Pull request by asn-d6**
This PR removes the functionality related to `COMPUTE_INDIVIDUAL_PROOFS`. After this PR, we will always do FK20, even if we are missing a single proof.

The rationale for this, is that it's unclear whether the `COMPUTE_INDIVIDUAL_PROOFS` optimization is fruitful, and it adds a moderate amount of complexity to the codebase.

For comparison, doing a full FK20 takes 200ms, wheras computing a single proof takes 50ms. Furthermore, the scenarios were one or two proofs are missing do not seem to be very frequent (based on our speculation), and even if that happens spending 200ms is likely OK.

If we ever receive reports of one or two proofs missing becoming a common pattern and that performance should be improved for those specific cases, we should consider re-adding this code.

An alternative would be to abstract away this functionality into its own function, and disable it with a compilation flag (as Justin has done right now). However, it seems hard to abstract this nicely, and right now it's inlined inside the main recovery function.

### CODE REVIEW

**Review by b-wagn — COMMENTED** [review]
This seems like a reasonable change to me.
I agree that removing functionality that is not used is a good idea to keep the codebase clean. 

Maybe what we should do at some point is add an explicit disclaimer to the readme that this is *not* a general purpose KZG library.

**Review by jtraglia — COMMENTED** [review]
LGTM. One little nit.

File: src/c_kzg_4844.c [hunk 1]
```diff
@@ -3722,9 +3493,6 @@ C_KZG_RET recover_cells_and_kzg_proofs(
                 recovered_proofs_g1, sizeof(g1_t), CELLS_PER_EXT_BLOB
             );
             if (ret != C_KZG_OK) goto out;
```
[the remark this answers is not in the document]
Comment (asn-d6) [reply src/c_kzg_4844.c]: Addressed at 52cb28b7c

**Review by jtraglia — APPROVED** [review]
LGTM, thanks!

**Pull request merged**